File: MyAnimeList/src/main/java/com/example/myanimelist/service/txt/TxtBackup.java
package com.example.myanimelist.service.txt;

import com.example.myanimelist.dto.LoadDTO;
import com.example.myanimelist.managers.DependenciesManager;
import com.example.myanimelist.utils.Properties;
import com.google.gson.Gson;

import java.io.*;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Collections;
import java.util.Optional;

public class TxtBackup implements ITxtStorage {

    private final Gson gson = DependenciesManager.getGson();

    public TxtBackup(){
        mkdir();
    }

    @Override
    public void mkdir() {
        Path dir = Path.of(Properties.JSON_DIR);
        if (Files.exists(dir)) return;
        try {
            Files.createDirectory(dir);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void save(LoadDTO dto) {
        try (var writer = new FileWriter(Properties.LOAD_FILE)) {
            gson.toJson(dto, writer);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public Optional<LoadDTO> load() {
        try (var reader = new FileReader(Properties.LOAD_FILE)) {
            return Optional.of(gson.fromJson(reader, LoadDTO.class));
        } catch (Exception e) {
            e.printStackTrace();
        }
        return Optional.empty();
    }

    public void changeNightMode(){
        String path = Properties.LOAD_FILE;
        String content = "";
        try{
            content = Files.readString(Path.of(path));
        } catch (IOException e){
            System.out.println("Error file cant be opened");
        }

        if(content.contains("\"nightMode\": false")){
            content = content.replace("\"nightMode\": false", "\"nightMode\": true");
        } else if(content.contains("\"nightMode\": true")) {
            content = content.replace("\"nightMode\": true", "\"nightMode\": false");
        }

        try(FileWriter fw = new FileWriter(path)){
            fw.write(content);
        } catch (IOException e){
            System.out.println("Error file cant be edited");
        }
    }
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/service/txt/TxtBackup.kt
package com.example.myanimelist.service.txt

import com.example.myanimelist.dto.LoadDTO
import com.example.myanimelist.utils.Properties
import com.example.myanimelist.utils.Themes
import com.google.gson.Gson
import org.koin.core.component.KoinComponent
import org.koin.core.component.inject
import java.io.FileReader
import java.io.FileWriter
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Path

class TxtBackup : ITxtStorage, KoinComponent {
    private val gson: Gson by inject()

    init {
        mkdir()
    }

    override fun mkdir() {
        val dir = Path.of(Properties.JSON_DIR)
        if (Files.exists(dir)) return
        try {
            Files.createDirectory(dir)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    override fun save(dtoList: LoadDTO) {
        try {
            FileWriter(Properties.LOAD_FILE).use { writer -> gson.toJson(dtoList, writer) }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun load(): LoadDTO? {
        try {
            FileReader(Properties.LOAD_FILE).use {
                return gson.fromJson(
                    it,
                    LoadDTO::class.java

                )
            }
        } catch (e: java.lang.Exception) {
            e.printStackTrace()
        }
        return null
    }

    companion object {
        fun changeNightMode(theme: Themes) {
            val path = Properties.LOAD_FILE
            var content = ""
            try {
                content = Files.readString(Path.of(path))
            } catch (e: IOException) {
                println("Error file cant be opened")
            }
            if (theme === Themes.OSCURO) {
                content = content.replace("\"nightMode\": false", "\"nightMode\": true")
            } else if (theme === Themes.CLARO) {
                content = content.replace("\"nightMode\": true", "\"nightMode\": false")
            }
            try {
                FileWriter(path).use { fw -> fw.write(content) }
            } catch (e: IOException) {
                println("Error file cant be edited")
            }
        }
    }
}